### ISSUE TITLE
EventListener error on correct identical function parameter

### ISSUE THREAD

**Issue by exapsy** [microsoft/TypeScript#35434]
**TypeScript Version:**  3.7.2, 3.5.1, 3.6.3, 3.4.5

**Search Terms:** Event Listener, addEventListener

**Code**

```ts
document.addEventListener('mousedown', onClick, false)

const onClick = (w: Window, e: MouseEvent): any => {
    // Code here
}
```

**Expected behavior:**
If the parameter types are the same, then no error should be emitted unless the syntax is incorrect.

**Actual behavior:**
Unless the names of the **parameters** are the same, an error is emitted saying that the function is not compatible.
![image](https://user-images.githubusercontent.com/9019120/69917817-e04a1480-1472-11ea-99bf-c46c80c72513.png)

[**Playground Link**](http://www.typescriptlang.org/play/?ts=3.5.1&ssl=1&ssc=1&pln=2&pc=22#code/CYewxgrgtgpgdgFwHQENjAKIDd4IDICWAzgvDAE4AUA5FCBETKAO5zUA0ABCHAMIA2BMAGsuAMxT9GASgBQssDxLc+gkZwC8nSswBcnAOoE4LLjH0BZeo2y5p+lHACemgHycA3rM4-OAej9OXhBgGE4ACwoYWQBfIA) 

**Related Issues:** #28357

**Comment by MartinJohns** [comment]
This is not a bug, but intended behavior. The `this` argument is a special kind of argument that is not passed along when calling the function, but instead describes the type of `this` instance.

The documentation should clear things up for you: http://www.typescriptlang.org/docs/handbook/functions.html#this-parameters-in-callbacks

**Comment by exapsy** [comment]
@MartinJohns Good to know. At least now if someone searches about the issue it will come up more easily :D 
**But** there's a big issue in any case (I'll make a seperate issue if needed). 
[Playground link](http://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=5&pc=55#code/MYewdgzgLgBODCAbAlsA1jAvDAFAdwC4YBZEAVwgFMBRAN0rCgEoiBDMATywD4YBvAFAxhMAPSiY8EABNKMABaUATpQEBfAQOkhgZALYMoAOlbTpdQwBlk0BspwByPeSra8YBwBo4YJKjTeAGasiFRMQA)

Obviously here type `this: Window` shouldn't match `MouseEvent` parameter type. It drew to a weird bug in my code.

**Comment by MartinJohns** [comment]
It's acceptable to omit the `this` parameter when it's not needed. It just means you can't access `this` type-safely in your function. But you can not provide a `this` parameter with an incompatible type, that will produce a compilation error.

**Comment by exapsy** [comment]
I see, that was super useful, thanx @MartinJohns , and the bug must have came from another source. 
Closing the issue.
